src/test/java/io/pinecone/PineconeIndexOperationsTest.java

```diff
@@ -238,5 +243,59 @@ public void testConfigureIndex() throws IOException {
         assertEquals(expectedConfiguredIndex, configuredIndex);
         assertEquals(requestCaptor.getValue().method(), "PATCH");
         assertEquals(requestCaptor.getValue().url().toString(), "https://api.pinecone.io/indexes/testIndex");
+
+        // Test for empty string for index name
+        PineconeException thrownEmptyIndexName = assertThrows(PineconeException.class, () -> client.configureIndex("",
+                configureIndexRequest));
+        assertEquals("Index name cannot be null or empty", thrownEmptyIndexName.getMessage());
+
+        // Test for null as index name
+        PineconeException thrownNullIndexName = assertThrows(PineconeException.class, () -> client.configureIndex(null,
+                configureIndexRequest));
+        assertEquals("Index name cannot be null or empty", thrownNullIndexName.getMessage());
+
+        // Test for null as configureIndexRequest
+        PineconeException thrownNullRequestObj = assertThrows(PineconeException.class,
+                () -> client.configureIndex("testIndex", null));
+        assertEquals("ConfigureIndexRequest object cannot be null", thrownNullRequestObj.getMessage());
+    }
+
+    @Test
+    public void testCreateCollection() throws IOException {
+        String filePath = "src/test/resources/collectionCreation.json";
+        String JsonStringCollection = new String(Files.readAllBytes(Paths.get(filePath)));
+        CollectionModel expectedCollection = gson.fromJson(JsonStringCollection, CollectionModel.class);
+
+        CreateCollectionRequest createCollectionRequest = new CreateCollectionRequest();
+        createCollectionRequest.setName("testCollection");
```

Comment:
Rather than setting the name in the test itself you could rely on what it is in the JSON fixture and keep things a bit more coupled. Currently the name `testCollection` and `test-collection` differ, which doesn't matter for this specific test, but may be confusing when reading the test.

Something like this:
```suggestion
        createCollectionRequest.setName(expectedCollection.getName());
```

Reply:
Weirdly it looks like I don't even need that at all in the test...